src/components/attendance/DataTableModal.tsx

```diff
@@ -58,6 +62,30 @@ const Body: React.FC<{ data: BodyProps | null | undefined }> = ({ data }) => {
   dayjs.extend(LocalizedFormat)
   const startDate = dayjs(data.startDate).format('lll')
   const endDate = dayjs(data.endDate).format('lll')
+  const modal = useContext(EventModalContext)
+  const confirmDelete = trpc.event.deleteEvent.useQuery(modal.id)
+  const router = useRouter()
+  
+  // const toast = useToast()
+  // const confirmDelete = async() => {
```

Comment:
how come this is commented out? You're going on the right track here using `mutateAsync` , but it'll require an input that you are passing in to know which event you are deleting. Along with the comment above, you could deconstruct what is returned by useMutation by doing `const { mutateAsync } = ... ` in order to get `mutateAsync` function.